File: Python/EstruturaDeRepeticao/exercicio_39.py
"""
Faça um programa que leia dez conjuntos de dois valores, o primeiro representando o número do aluno e o segundo representando a sua altura em centímetros.
Encontre o aluno mais alto e o mais baixo.
Mostre o número do aluno mais alto e o número do aluno mais baixo, junto com suas alturas.
"""

numero_alunos = 10
numeros_alunos = []
alturas_alunos = []

for i in range(numero_alunos):
    numero_aluno = int(input(f"Digite o número do aluno {i + 1}: "))
    altura_aluno = int(input(f"Digite a altura do aluno {i + 1} (em cm): "))

    numeros_alunos.append(numero_aluno)
    alturas_alunos.append(altura_aluno)

indice_aluno_mais_alto = alturas_alunos.index(max(alturas_alunos))
indice_aluno_mais_baixo = alturas_alunos.index(min(alturas_alunos))

numero_aluno_mais_alto = numeros_alunos[indice_aluno_mais_alto]
numero_aluno_mais_baixo = numeros_alunos[indice_aluno_mais_baixo]
altura_mais_alto = alturas_alunos[indice_aluno_mais_alto]
altura_mais_baixo = alturas_alunos[indice_aluno_mais_baixo]

print(f"Aluno mais alto - Número: {numero_aluno_mais_alto}, Altura: {altura_mais_alto}cm")
print(f"Aluno mais baixo - Número: {numero_aluno_mais_baixo}, Altura: {altura_mais_baixo}cm")

File: Java/EstruturaDeRepeticao/Exercicio39.java
// Faça um programa que leia dez conjuntos de dois valores, o primeiro representando o número do aluno e o segundo representando a sua altura em centímetros.
// Encontre o aluno mais alto e o mais baixo.
// Mostre o número do aluno mais alto e o número do aluno mais baixo, junto com suas alturas.
package EstruturaDeRepeticao;
import java.util.Scanner;

public class Exercicio39 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int numeroAlunoMaisAlto = 0, numeroAlunoMaisBaixo = 0;
        int alturaMaisAlto = 0, alturaMaisBaixo = Integer.MAX_VALUE;

        for (int i = 1; i <= 10; i++) {
            System.out.print("Digite o número do aluno " + i + ": ");
            int numeroAluno = scanner.nextInt();
            System.out.print("Digite a altura do aluno (em centímetros): ");
            int alturaAluno = scanner.nextInt();

            if (alturaAluno > alturaMaisAlto) {
                alturaMaisAlto = alturaAluno;
                numeroAlunoMaisAlto = numeroAluno;
            }

            if (alturaAluno < alturaMaisBaixo) {
                alturaMaisBaixo = alturaAluno;
                numeroAlunoMaisBaixo = numeroAluno;
            }
        }

        System.out.println("Aluno mais alto:");
        System.out.println("Número do aluno: " + numeroAlunoMaisAlto);
        System.out.println("Altura: " + alturaMaisAlto + " cm");

        System.out.println("Aluno mais baixo:");
        System.out.println("Número do aluno: " + numeroAlunoMaisBaixo);
        System.out.println("Altura: " + alturaMaisBaixo + " cm");

        scanner.close();
    }
}
